micras_hal/include/micras/hal/pwm.hpp

```diff
@@ -24,34 +24,39 @@ class Pwm {
     };
 
     /**
-     * @brief Construct a new Pwm object
+     * @brief Construct a new Pwm object.
      *
-     * @param config Configuration for the PWM
+     * @param config Configuration for the PWM.
      */
     explicit Pwm(const Config& config);
 
     /**
-     * @brief Set the PWM duty cycle
+     * @brief Set the PWM duty cycle.
      *
-     * @param duty_cycle Duty cycle value
+     * @param duty_cycle Duty cycle value.
      */
     void set_duty_cycle(float duty_cycle);
 
     /**
-     * @brief Set the PWM frequency
+     * @brief Set the PWM frequency.
      *
-     * @param frequency Frequency value in Hz
+     * @note Changing the pwm frequency will also change the duty cycle and reset the counter.
```

Comment:
n sei se `will also change the duty cycle` eh descritivo o suficiente. Se eu ler rapido no meio do debug posso inclusive assumir que "ah ok, ele ja resolve isso pra mim"